packages/bsky/tests/views/author-feed.test.ts

```diff
@@ -232,4 +235,46 @@ describe('pds author feed views', () => {
       },
     )
   })
+
+  it('can filter by type', async () => {
+    const { data: allFeed } = await agent.api.app.bsky.feed.getAuthorFeed({
+      actor: carol,
+    })
+
+    expect(
+      allFeed.feed.some(({ post }) => {
+        return isRecord(post.record) && Boolean(post.record.reply)
+      }),
+    ).toBeTruthy()
+    expect(
+      allFeed.feed.some(({ post }) => {
+        return (
+          isEmbedRecordWithMedia(post.embed) && isImageEmbed(post.embed?.media)
+        )
+      }),
+    ).toBeTruthy()
+
+    const { data: mediaFeed } = await agent.api.app.bsky.feed.getAuthorFeed({
+      actor: carol,
+      filter: 'posts_with_media',
+    })
+
+    expect(
+      mediaFeed.feed.every(({ post }) => {
+        return (
+          isEmbedRecordWithMedia(post.embed) && isImageEmbed(post.embed?.media)
```

Comment:
ah good call on the `&&` here 👍 

and carol doesn't have any posts with just an "image embed", only posts with a "record with media embed"
but just to make the tests a bit more self documenting, I think this should be 

```
(isEmbedRecordWithMedia(post.embed) && isImageEmbed(post.embed?.media)) || isImageEmbed(post.embed)
```

Reply:
the embeds logic is a bit heady so lmk if you want me to walk you through it at all

Reply:
Ok I messed around with some more seed data and I think I grok this. Pry wouldn't hurt to walk through it some time but no rush. Updated this assertion, as well as added an images only post for Carol (and updated the snapshot).